Packages/com.unity.inputsystem/InputSystem/Plugins/WebGL/WebGLSupport.cs

```diff
@@ -10,22 +14,130 @@ namespace UnityEngine.InputSystem.WebGL
 #endif
     static class WebGLSupport
     {
+        private const string InterfaceName = "WebGL";
         public static void Initialize()
         {
             // We only turn gamepads with the "standard" mapping into actual Gamepads.
             InputSystem.RegisterLayout<WebGLGamepad>(
                 matches: new InputDeviceMatcher()
-                    .WithInterface("WebGL")
+                    .WithInterface(InterfaceName)
                     .WithDeviceClass("Gamepad")
                     .WithCapability("mapping", "standard"));
 
-            ////TODO: add a layout builder for this that dynamically looks at the axis and button count
-            // For all other WebGL "gamepads", we don't know the button and axis mappings so we turn
-            // them into joysticks.
-            InputSystem.RegisterLayout<WebGLJoystick>(
-                matches: new InputDeviceMatcher()
-                    .WithInterface("WebGL")
-                    .WithDeviceClass("Gamepad"));
+            InputSystem.onFindLayoutForDevice += OnFindLayoutForDevice;
+        }
+
+        internal static string OnFindLayoutForDevice(int deviceId, ref InputDeviceDescription description,
+            string matchedLayout, IInputRuntime runtime)
+        {
+            // If the device isn't a WebGL device, we're not interested.
+            if (string.Compare(description.interfaceName, InterfaceName, StringComparison.InvariantCultureIgnoreCase) != 0)
+                return null;
+
+            // If it was matched by the standard mapping, we don't need to fall back to generating a layout.
+            if (matchedLayout == typeof(WebGLGamepad).Name)
```

Comment:
I think 

```
if (!string.IsNullOrEmpty(matchedLayout))
```

would be a better solution. This way we can go and add support for other WebGL gamepads/joysticks without this breaking.

Reply:
I tried that first, but it does not work, because all WebGL devices are matched by a generic "Gamepad" layout otherwise. Any idea why?

Reply:
Argh, that is annoying. Didn't consider that. But maybe 

```
if (!string.IsNullOrEmpty(matchedLayout) && matchedLayout != "Gamepad")
```

then?

Reply:
ok